scripts/download_openmeteo_2024_2025.py

```
@@ -0,0 +1,151 @@
+        except subprocess.TimeoutExpired:
+            print(f"  ✗ Download timeout for {month_name} (exceeded 1 hour)")
+            process.kill()
+            failed.append(month_name)
+            
+        except KeyboardInterrupt:
+            print("\n\n⚠️  Download interrupted by user")
+            print(f"  Completed: {len(successful)} months")
+            print(f"  Remaining: {len(months_to_download) - i} months")
+            return 1
+            
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)